agent/fw_drupal.c

```diff
@@ -454,25 +454,40 @@ static void nr_drupal_wrap_hook_within_module_invoke_all(
    * module_invoke_all(), the drupal_module_invoke_all_hook global should be
    * available.
    */
-  if (NULL == NRPRG(drupal_module_invoke_all_hook)) {
+#if ZEND_MODULE_API_NO >= ZEND_8_0_X_API_NO \
+     && !defined OVERWRITE_ZEND_EXECUTE_DATA
+  zval* curr_hook = (zval*)nr_stack_get_top(&NRPRG(drupal_module_invoke_all_hooks));
```

Comment:
Is it possible to add unit tests that exercise the functionality of the new hooks stack?
https://github.com/newrelic/newrelic-php-agent/blob/main/agent/tests/test_fw_drupal.c